src/events/events.service.ts

```diff
@@ -596,46 +596,36 @@ export class EventsService {
   }
 
   async updateLatestPoints(userId: number, type: EventType): Promise<void> {
-    const occurredAtAggregate = await this.prisma.readClient.event.aggregate({
-      _max: {
-        occurred_at: true,
-      },
-      where: {
-        type,
-        user_id: userId,
-      },
-    });
-    const latestOccurredAt = occurredAtAggregate._max.occurred_at;
-
-    const pointsAggregate = await this.prisma.readClient.event.aggregate({
-      _sum: {
-        points: true,
-      },
-      _count: {
-        points: true,
-      },
-      where: {
-        type,
-        user_id: userId,
-      },
-    });
-    const points = pointsAggregate._sum.points ?? 0;
-    const count = pointsAggregate._count.points ?? 0;
-
-    const totalPointsAggregate = await this.prisma.readClient.event.aggregate({
-      _sum: {
-        points: true,
-      },
-      where: {
-        user_id: userId,
-      },
-    });
-    const totalPoints = totalPointsAggregate._sum.points ?? 0;
+    const [counts] = await this.prisma.readClient.$queryRawUnsafe<
+      {
+        count: BigInt;
```

Comment:
Looks like you are only using 4 of the 5 selected values? Also I take it ordering doesn't matter?

Reply:
yeah, order of keys in JS doesn't matter.

Reply:
Removed total_count.